### PR TITLE
Fix: update Tendermint tests and validator logic

### PR DESCRIPTION
**Pull request by rianhughes**
closes https://github.com/NethermindEth/juno/issues/2894

In Tendermint, the proposer only sends a subset of the gas fees we need to compute the block hash. We should use the header values by default. This prevents validator_test.go tests from crashing when we will update supportedStarknetVersion to 0.14.0.

Also, I updated the empty flow test to use the same custom blockchain as the non-empty flow test. There's no need to have both.

Also also, as discussed with @infrmtcs  we require tendermint to run on the latest starknet versions, not old versions. The reason is the old blocks don't have the values we need to compute the blockhash function (namely the fee prices)

### CODE REVIEW

**Review by rodrigo-pino — APPROVED** [review]
Approving to unblock

File: consensus/tendermint/tendermint.go [hunk 1]
```diff
@@ -8,8 +8,14 @@ import (
 	"github.com/NethermindEth/juno/utils"
 )
 
+// H 3 R3 , we propose
+// Call Value() V - this blocks out consensus progress
+// The network commits H3 R2 while we generate a proposal
+// We could cancel the block builder, and move to the new height
+// Msg times : delta + maxTime
+
```
Comment (infrmtcs) [inline consensus/tendermint/tendermint.go]: Don't forget to remove our discussion draft :smile:

**Review by infrmtcs — APPROVED** [review]
LGTM, just need to remove the unrelated comments

File: consensus/tendermint/tendermint.go (hunk 1 above)
Comment (rianhughes) [reply consensus/tendermint/tendermint.go]: Lol nice catch!

Comment (rianhughes) [reply consensus/tendermint/tendermint.go]: Done
